src/transformers/modeling_utils.py

```diff
@@ -818,7 +818,6 @@ def _move_model_to_meta(model, loaded_state_dict_keys, start_prefix):
 def _load_state_dict_into_meta_model(
     model,
     state_dict,
-    loaded_state_dict_keys,  # left for now but could be removed, see below
```

Comment:
Here, I removed `loaded_state_dict_keys` from `_load_state_dict_into_meta_model`, because according to the following snippet, it was not actually used before:
```py
        # First part of the test is always true as load_state_dict_keys always contains state_dict keys.
        if param_name not in loaded_state_dict_keys or param_name not in expected_keys:
```

I might have overlooked some downside effects, especially with quantization and/or training frameworks. WDYT @ArthurZucker and @LysandreJik ? Who should I tag for more info?

Also happy to change back to the original behaviour

Reply:
If it doesn't break any tests, let's remove it and keep an eye out for eventual breakage